### ISSUE TITLE
MessagePack codec (or VRL functions)

### ISSUE THREAD

**Issue by micoq** [vectordotdev/vector#8675]
We have a Kafka topic with msgpack events we would like to deserialize (and reserialize after some transformations in an other topic).
Currently, we use Logstash for that with this plugin : https://github.com/logstash-plugins/logstash-codec-msgpack

It could be very useful to have a generic msgpack codec in Vector like for JSON format, maybe in the form of a pair VRL functions `parse_msgpack` / `encode_msgpack` ?

**Comment by jszwedko** [comment]
Agreed, thanks for the idea @micoq ! I think we can support it both as a codec on sources as well as a VRL function.

cc/ @pablosichert who has been working on improving codec support recently

**Comment by Brabalawuka** [comment]
Any plan to support message pack this year?

**Comment by keuin** [comment]
Hi @pront, I have a working MessagePack decoder in PR #24510. It's an analogue to JSON decoder. I can work on it and add an encoder to solve this feature request, if you want this PR to proceed. Or just close it and let me know.

**Comment by pront** [comment]
> Hi [@pront](https://github.com/pront), I have a working MessagePack decoder in PR [#24510](https://github.com/vectordotdev/vector/pull/24510). It's an analogue to JSON decoder. I can work on it and add an encoder to solve this feature request, if you want this PR to proceed. Or just close it and let me know.

Thanks for the heads-up, @keuin! 

Please feel free to continue working on it. Adding the encoder and getting to a full codec would be great. Just want to set expectations here: we have a significant PR backlog right now, so review may take some time. That said, the work won't go to waste. Feel free to ping here if you have questions.
